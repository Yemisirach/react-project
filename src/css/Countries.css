.flag-info {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 8px;
  background-color: #efe6e9;
  padding: 32px;
  text-align: center;
  background-color: #ee6b99;
}
.main-list {
  background-color: #ee6b99;
}
.selected-country-details h3 {
  color: #060606 !important;
  margin-bottom: 12px;
  padding: 8px;
}

ul {
  display: flex;
  flex-direction: column;
}
ul strong:nth-child(odd) {
  background-color: #bd2459;
}
ul strong:nth-child(even) {
  color: black;
}
.country-details-content {
  color: #fff;
  background-color: black;
  justify-content: center;
  width: 80%;
  margin: auto;
  border-radius: 5px;
}

strong {
  background-color: #ee6b99;
  padding: 1rem;
  margin: 0.1rem;
  border-radius: 5px;
  font-size: 15px;
}

.country-flag {
  height: 10rem;
  border: 1px;
  box-shadow: 0 0 15px 0 #070707;
  margin-bottom: 0.5rem;
}
.flag-image {
  background: #d7376f;
  padding: 3%;
  border-radius: 5px;
}
