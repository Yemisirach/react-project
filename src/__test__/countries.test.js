import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Store from '../redux/store';
import Countries from '../components/countries';

describe('testing countries components', () => {
  it('test countries component', () => {
    const data = render(
      <Provider store={Store}>
        <Countries />
      </Provider>,
    );
    expect(data).toMatchSnapshot();
  });
});
