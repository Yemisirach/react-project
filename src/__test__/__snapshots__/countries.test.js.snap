// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing countries components test countries component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-list"
      >
        <div
          class="flag-info"
        >
          <div
            class="flag-image"
          >
            <img
              class="country-flag"
            />
            <h2 />
          </div>
        </div>
        <div
          class="selected-country-details"
        >
          <h3
            class="country-details"
          >
            Stats Breakdown
          </h3>
        </div>
        <ul
          class="country-details-content"
        >
          <strong>
            Timezone:
          </strong>
          <strong>
            Capital:
          </strong>
          <strong>
            Population:
             
          </strong>
          <strong />
          <strong>
            Area:
             
             
            km
            <sup>
              2
            </sup>
          </strong>
          <strong>
            Languages:
          </strong>
          <strong>
            Currencies:
          </strong>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-list"
    >
      <div
        class="flag-info"
      >
        <div
          class="flag-image"
        >
          <img
            class="country-flag"
          />
          <h2 />
        </div>
      </div>
      <div
        class="selected-country-details"
      >
        <h3
          class="country-details"
        >
          Stats Breakdown
        </h3>
      </div>
      <ul
        class="country-details-content"
      >
        <strong>
          Timezone:
        </strong>
        <strong>
          Capital:
        </strong>
        <strong>
          Population:
           
        </strong>
        <strong />
        <strong>
          Area:
           
           
          km
          <sup>
            2
          </sup>
        </strong>
        <strong>
          Languages:
        </strong>
        <strong>
          Currencies:
        </strong>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
